Makefile

```diff
@@ -735,16 +749,18 @@ init-submodules-e: init-webapps-submodules-e
 	git submodule init e
 	git submodule update
 
+# Update go.mod and vendor files. 
 .PHONY: update-vendor
 update-vendor:
 	# update modules in api/
 	cd api && go mod tidy
 	# update modules in root directory
 	go mod tidy
 	go mod vendor
-	# delete the vendored api package. In its place
-	# create a symlink to the the original api package
-	rm -r vendor/github.com/gravitational/teleport/api/v7
+	# delete the vendored api package.
+	rm -rf vendor/github.com/gravitational/teleport/api
+	# create a symlink to the the original api package - /api or /api/vX if X > 1
```

Comment:
This is because the script has to adapt depending on the version. Now that I reverted the version back to 0, this looks like
```
if [ $(shell echo $(VERSION) | cut -d "." -f1) -gt 1 ]; then mkdir -p vendor/github.com/gravitational/teleport/api; fi
ln -s -r $(shell readlink -f api) vendor/github.com/gravitational/teleport/api
```

I moved the `vendor-api` logic to a separate target and added some better comments, let me know if it makes sense now.